containerd-http-proxy/configure_http_proxy.yaml

```
@@ -0,0 +1,107 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
+            input_string=$1
+
+            if echo "$input_string" | grep -qvE '^(HTTP_PROXY|HTTPS_PROXY|NO_PROXY)='; then
+                echo "Error: Input does not start with a valid prefix (HTTP_PROXY, HTTPS_PROXY, NO_PROXY). Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | grep -q ' '; then
+                echo "Error: Input cannot contain spaces. Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | sed 1d | grep -q .; then
+                echo "Error: Input cannot contain newline. Input: '$input_string'"
+                exit 1
+            fi
+          }
+
+          validate_proxy "${HTTP_PROXY:-}"
+          validate_proxy "${HTTPS_PROXY:-}"
+          validate_proxy "${NO_PROXY:-NO_PROXY=localhost}"
+
+          STARTUP_SCRIPT=$(cat <<END
+          set -e
+          set -u
+          mkdir -p /etc/systemd/system/containerd.service.d
+          echo "[Service]
+          Environment=${HTTP_PROXY}
```

`${HTTP_PROXY}` must be in quotes

```suggestion
          Environment="${HTTP_PROXY}"
```

---

oh, just realized it is even worse than that. We need quotes for bash AND quotes written into the file itself.

---

Added quote: Environment="HTTP_PROXY=${HTTP_PROXY}"